lib/compiler-singlepass/src/compiler.rs

```diff
@@ -113,7 +112,7 @@ impl Compiler for SinglepassCompiler {
 
                 while generator.has_control_frames() {
                     generator.set_srcloc(reader.original_position() as u32);
-                    let op = reader.read_operator().map_err(to_compile_error)?;
+                    let op = reader.read_operator()?;
```

Comment:
Before this PR, `BinaryReaderError` was converted to `CompileError::Codegen` at this place. Not it is converted to `CompileError::Wasm`. Is this change correct?

Reply:
Hmm, it's a fuzzy line for sure.  Perhaps the idea for calling it "codegen" was that we only read more operators here based on state of the code generator... whereas calling it a "wasm" error suggests that if the invariant here isn't met, then it's because of a problem with the Wasm...

I would defer to the types of errors that this function (and functions like it) are returning... if they're mostly returning codegen errors, then I'd expect that to be what we want here, likewise with Wasm

Reply:
Alright, `reader.read_operator()` now returns `WasmError`, so as long as you don't have a better idea, I leave it to the default type converter.

One thought I had about this code section: Maybe `to_compile_error` should be renamed to `to_codegen_error` if it should always produces `CompileError::Codegen`, independent of the source type.

Reply:
Hmm, it looks like `to_compile_error` comes from the trait / method, so it probably makes sense to keep it is as used if that's a widely used abstraction here -- I'm not super familiar with this part of wasmer's code